<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainAnimation</class>
 <widget class="QMainWindow" name="MainAnimation">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>LaunchPad LED Animator</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resource.qrc">
    <normaloff>:/icons/Icons/black/preview.png</normaloff>:/icons/Icons/black/preview.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QListWidget" name="activeColorList">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>48</width>
            <height>30</height>
           </size>
          </property>
          <property name="statusTip">
           <string>This is current color.</string>
          </property>
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="currentRow">
           <number>-1</number>
          </property>
          <item>
           <property name="text">
            <string/>
           </property>
           <property name="background">
            <brush brushstyle="SolidPattern">
             <color alpha="255">
              <red>0</red>
              <green>0</green>
              <blue>0</blue>
             </color>
            </brush>
           </property>
           <property name="flags">
            <set>ItemIsEnabled</set>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="colorList">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>48</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="statusTip">
           <string>Click to change current color.</string>
          </property>
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="currentRow">
           <number>-1</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGraphicsView" name="buttonGrid">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="statusTip">
         <string>Select rectangles to color them.</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QSpinBox" name="startEdit">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="maximum">
         <number>999</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="gotoStart">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="resource.qrc">
          <normaloff>:/icons/Icons/start.svg</normaloff>:/icons/Icons/start.svg</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="gotoPrev">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="resource.qrc">
          <normaloff>:/icons/Icons/prev.svg</normaloff>:/icons/Icons/prev.svg</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="keyButton">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="resource.qrc">
          <normaloff>:/icons/Icons/key-add.svg</normaloff>:/icons/Icons/key-add.svg</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="curEdit">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="maximum">
         <number>999</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="play">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="resource.qrc">
          <normaloff>:/icons/Icons/play.svg</normaloff>:/icons/Icons/play.svg</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="gotoNext">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="resource.qrc">
          <normaloff>:/icons/Icons/next.svg</normaloff>:/icons/Icons/next.svg</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="gotoEnd">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="resource.qrc">
          <normaloff>:/icons/Icons/end.svg</normaloff>:/icons/Icons/end.svg</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QSpinBox" name="endEdit">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="maximum">
         <number>999</number>
        </property>
        <property name="value">
         <number>100</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QSlider" name="frameSlider">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="action_Open"/>
    <addaction name="action_Save"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menu_Device">
    <property name="title">
     <string>&amp;Device</string>
    </property>
    <addaction name="actionSend"/>
    <addaction name="separator"/>
    <addaction name="actionLive_Preview"/>
    <addaction name="actionPlay_Indifinitly"/>
    <addaction name="separator"/>
    <addaction name="actionSet_Trigger"/>
    <addaction name="actionListen"/>
   </widget>
   <widget class="QMenu" name="menuPad">
    <property name="title">
     <string>Pad</string>
    </property>
    <addaction name="actionClear"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menu_Device"/>
   <addaction name="menuPad"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="action_Open"/>
   <addaction name="action_Save"/>
   <addaction name="separator"/>
   <addaction name="actionClear"/>
   <addaction name="actionLive_Preview"/>
   <addaction name="actionPlay_Indifinitly"/>
   <addaction name="actionSend"/>
   <addaction name="separator"/>
   <addaction name="actionSet_Trigger"/>
   <addaction name="actionListen"/>
   <addaction name="separator"/>
  </widget>
  <action name="action_Open">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/open.svg</normaloff>:/icons/Icons/open.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Open...</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/save.svg</normaloff>:/icons/Icons/save.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Save...</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/save.svg</normaloff>:/icons/Icons/save.svg</iconset>
   </property>
   <property name="text">
    <string>Save As...</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/exit.svg</normaloff>:/icons/Icons/exit.svg</iconset>
   </property>
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="actionSend">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/send.svg</normaloff>:/icons/Icons/send.svg</iconset>
   </property>
   <property name="text">
    <string>Send Current View</string>
   </property>
  </action>
  <action name="actionSend_Animation">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/sendall.svg</normaloff>:/icons/Icons/sendall.svg</iconset>
   </property>
   <property name="text">
    <string>Send Animation</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/clear.svg</normaloff>:/icons/Icons/clear.svg</iconset>
   </property>
   <property name="text">
    <string>Clear</string>
   </property>
  </action>
  <action name="actionLive_Preview">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/preview.svg</normaloff>:/icons/Icons/preview.svg</iconset>
   </property>
   <property name="text">
    <string>Live Preview</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/new.svg</normaloff>:/icons/Icons/new.svg</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/about.svg</normaloff>:/icons/Icons/about.svg</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionListen">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/listen.svg</normaloff>:/icons/Icons/listen.svg</iconset>
   </property>
   <property name="text">
    <string>Listen</string>
   </property>
  </action>
  <action name="actionSet_Trigger">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/trigger.svg</normaloff>:/icons/Icons/trigger.svg</iconset>
   </property>
   <property name="text">
    <string>Set Trigger</string>
   </property>
   <property name="statusTip">
    <string>Click on a rectangle to set a trigger.</string>
   </property>
  </action>
  <action name="actionPlay_Indifinitly">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/icons/Icons/loop.svg</normaloff>:/icons/Icons/loop.svg</iconset>
   </property>
   <property name="text">
    <string>Play Indifinitly</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>curEdit</tabstop>
  <tabstop>startEdit</tabstop>
  <tabstop>endEdit</tabstop>
  <tabstop>play</tabstop>
  <tabstop>keyButton</tabstop>
  <tabstop>gotoPrev</tabstop>
  <tabstop>gotoStart</tabstop>
  <tabstop>gotoNext</tabstop>
  <tabstop>gotoEnd</tabstop>
  <tabstop>frameSlider</tabstop>
  <tabstop>buttonGrid</tabstop>
  <tabstop>activeColorList</tabstop>
  <tabstop>colorList</tabstop>
 </tabstops>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>curEdit</sender>
   <signal>valueChanged(int)</signal>
   <receiver>frameSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>589</x>
     <y>719</y>
    </hint>
    <hint type="destinationlabel">
     <x>583</x>
     <y>747</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frameSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>curEdit</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>562</x>
     <y>750</y>
    </hint>
    <hint type="destinationlabel">
     <x>570</x>
     <y>721</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
